bark/tools/genrasters.py

```diff
@@ -0,0 +1,121 @@
+import argparse
+import bark
+import collections as coll
+import matplotlib.pyplot as plt
+import numpy as np
+import os
+import scipy.io.wavfile
+import sys
+
+# plot styling constants
+AXIS_LABEL_SIZE = 12 # in pt
+TITLE_SIZE = 16 # in pt
+OFFSET_STEP = 1 # arbitrary units
+RASTER_LH_PROP = 1 # line height proportion
+RASTER_LW = 1 # line width; in pt
+STIM_OFFSET = 1.1 # proportion of OFFSET_STEP
+DFLT_PAD_BEF = 0.5 # in seconds
+DFLT_PAD_AFT = 0.5 # in seconds
+
+Stimulus = coll.namedtuple('Stimulus', ['name', 'data', 'sampling_rate'])
+
+def sec2samp(sec, sr):
+    return int(round(sec * sr))
+
+def pad_stimulus(stim, padding):
+    # stim     --  Stimulus object
+    # padding  --  2-tuple, in seconds, of before and after padding
+    # returns a new Stimulus object with appropriate before and after padding
+    extended = [0]*sec2samp(padding[0], stim.sampling_rate)
+    extended.extend(stim.data)
+    extended.extend([0]*sec2samp(padding[1], stim.sampling_rate))
+    return Stimulus(stim.name, extended, stim.sampling_rate)
+
+def aligned_raster(spikes, stim_ds, stim_name, padding, title, stim_data):
+    # spike_list  --  sequence of spike times, in seconds
```

Comment:
'#' for docstrings doesn't work. Use triple quotes instead. See https://www.python.org/dev/peps/pep-0257/#specification